provider/core/src/resource.rs

```diff
@@ -39,13 +39,14 @@ impl ResourceKeyHash {
 /// const K: ResourceKey = icu_provider::resource_key!("foo/bar@1");
 /// ```
 ///
-/// The path string has to contain at least one `/`, and end with `@` followed by one or more ASCII
-/// digits. Paths do not contain characters other than ASCII letters and digits, `_`, `/`, `=`, and
-/// `@`. Invalid paths are compile-time errors (as [`resource_key!`] uses `const`).
+/// The human-readable path string ends with `@` followed by one or more digits (the version
```

Comment:
Okay.  I kind-of prefer requiring a `/` because it enforces namespacing, but this is a soft constraint and not something that needs to be in the syntax, I suppose.

Reply:
I agree that namespaces are a style choice and not a technical requirement. If a user only uses their own keys we shouldn't force that on them.
 
I wouldn't really consider the current first parts of keys like `list/and@1` namespaces (list is somewhat special because the `and` is more like version, the base key is really just `list`). Anyway, we don't "own" the term `list`, so if we wanted namespaces to avoid clashing with externally defined keys, we should use something like `icu/`.